lib/hooks.ts

```diff
@@ -0,0 +1,25 @@
+import { useState, useEffect } from 'react'
+
+/** Listens to changes in window size and returns the width. Ref: https://blog.logrocket.com/developing-responsive-layouts-with-react-hooks/ */
+const useViewport = () => {
```

Comment:
Will need to use this custom hook for retrieving viewports in the Map component too. Not relevant to this pr, but thought it would be good to inform and keep in mind so we can do that later.